plugin/src/main/java/org/opensearch/ml/helper/ConnectorAccessControlHelper.java

```diff
@@ -70,13 +79,49 @@ public void validateConnectorAccess(Client client, String connectorId, ActionLis
             getConnector(client, connectorId, ActionListener.wrap(connector -> {
                 boolean hasPermission = hasPermission(user, connector);
                 wrappedListener.onResponse(hasPermission);
-            }, e -> { wrappedListener.onFailure(e); }));
+            }, wrappedListener::onFailure));
         } catch (Exception e) {
             log.error("Failed to validate Access for connector:" + connectorId, e);
             listener.onFailure(e);
         }
     }
 
+    public void validateConnectorAccess(
+        SdkClient sdkClient,
+        Client client,
```

Comment:
Why do we require `Client`? all operations can be done using `SDKClient`

Reply:
Refer to this comment: https://github.com/opensearch-project/ml-commons/pull/2459#discussion_r1607312600